query_optimizer/logical/Filter.hpp

```diff
@@ -71,6 +74,9 @@ class Filter : public Logical {
     return Filter::Create(new_children[0], filter_predicate_);
   }
 
+  LogicalPtr copyWithNewInputExpressions(
+      const std::vector<expressions::ExpressionPtr>& input_expressions) const override;
```

Comment:
If we apply the reference sigil standard here, it should look like:

`const std::vector<expressions::ExpressionPtr> &input_expressions) const override;`